src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_BuildSynchronization.cs

```diff
@@ -88,8 +95,21 @@ public async Task SynchronizeWithBuildAsync(ImmutableDictionary<ProjectId, Immut
                         AnalyzerService.Reanalyze(Workspace, documentIds: ImmutableArray.Create(activeDocument.Id), highPriority: true);
                     }
 
-                    // Enqueue re-analysis of open documents.
-                    AnalyzerService.Reanalyze(Workspace, documentIds: Workspace.GetOpenDocumentIds(), highPriority: true);
+                    // Enqueue remaining re-analysis on the task queue.
```

Comment:
needs explanation of why we do this here.

as an aside, i struggle a lot having any idea what the data/contorl flow is here with these differnet queues and differnet things happening all over the place.